File: app/src/main/java/com/astune/mcenter/object/Link/NewLink.java
package com.astune.mcenter.object.Link;

import com.astune.mcenter.utils.enums.LinkType;

public class NewLink implements Link {
    private final String name;
    private final int parentId;
    private final LinkType type = LinkType.NEW_LINK;

    public NewLink(int parentId, String name){
        this.parentId = parentId;
        this.name = name;
    }
    @Override
    public String getName() {
        return name;
    }

    @Override
    public int getParentId() {
        return parentId;
    }

    @Override
    public LinkType getType() {
        return type;
    }

    @Override
    public String getInfo() {
        return "";
    }
}

File: core/database/src/main/java/com/astune/database/Link/NewLink.kt
package com.astune.database.Link

import com.astune.model.LinkType

/**
 * the Link Insertion btn
 */
class NewLink(override val name: String, override val parent: Int, override val id: Int = -1) : Link {
    override val type = LinkType.NEW_LINK
    override val info: String
        get() = ""
}

class EmptyLink(override val name: String = "", override val parent: Int = 0, override val id: Int = 0) : Link {
    override val type = LinkType.EMPTY_LINK
    override val info: String
        get() = ""
}
